app/controllers/newsletter_signups_controller.rb

```diff
@@ -0,0 +1,34 @@
+class NewsletterSignupsController < ApplicationController
+  before_action :set_newsletter_signup
+
+  def create
+    signup = CreateNewsletterSignup.new(
+      {
+        first_name: @newsletter_signup[:first_name],
+        last_name: @newsletter_signup[:last_name],
+        email: @newsletter_signup[:email]
+      }
+    ).perform
+
+    if signup.class == SendGrid::Response
+      response = JSON.parse(signup.body)
+      if response['errors']
+        redirect_back(fallback_location: root_path, error: response['errors'][0]['message'])
+      else
+        redirect_back(fallback_location: root_path, success: 'Thank you for signing up!')
+      end
+    else
+      redirect_back(fallback_location: root_path, error: signup.message)
+    end
+  end
+
+  private
+
+  def set_newsletter_signup
+    @newsletter_signup = params[:newsletter_signup]
+  end
+
```

Comment:
suggestion: no need to set an @var here, just reference `newsletter_signup_params[:thing]` where needed